### DISCUSSION TITLE
Can't unequip option

### DISCUSSION THREAD

**Discussion by da-rulez** [AscensionGameDev/Intersect-Engine#2183] [Ideas]
A new option to disallow players to unequip an item once equipped. The only way to unequip would be through the event system. 

It's nice in minigames, for quest tasks or even as punishment.

Generic enough that anyone can use it in their game, and easy enough to ignore if you don't need it. 

EDIT: 

After some more thinking, in some cases you would want to be able to get rid of the item. For example if you get stuck in a puzzle. Would be good to add an optional 'destroy' in the right click menu for forced equipped items.

**Comment by PyroTech03** [comment]
Should set an option for if unequip it just drops item. As currently you can't drop equipped items.

**Reply by da-rulez** [reply]
Yeah or destroy possibly.

**Reply by PyroTech03** [reply]
Both would be ideal. For me, it would allow simulating carrying a large object. Have item take weapon slot 2 hand. If they drop item shouldn't reset.

**Comment by Arufonsu** [comment]
1- A destroy dropdown option should be there only if allowed to be destroyed by the server / item editor - and is quite not related to this topic imo
2- A `locked when equipped` checkbox within the item editor is useful to have proper comestics such as hairs/eye colors/etc, you don't want a "destroy" button on those, you don't want players to be able to take these off (normally) - what should be implemented along is a checkbox to bypass the locked-equipment thru the unequip event, similar to what we did by adding a checkbox to bypass bonded spells within the spell removal event. this way, things like cosmetics/hairs/eye colors/ could be easily handled by event dialogues by unequipping/removing (+ the bypass checkbox)  and adding/equipping items with the flag `locked when equipped`
